flink-core/src/test/java/org/apache/flink/util/NetUtilsTest.java

```diff
@@ -61,25 +56,45 @@ public void testParseHostPortAddress() {
     @Test
     public void testAcceptWithoutTimeout() throws IOException {
         // Validates that acceptWithoutTimeout suppresses all SocketTimeoutExceptions
-        ServerSocket serverSocket = mock(ServerSocket.class);
-        when(serverSocket.accept())
-                .thenAnswer(
-                        new Answer<Socket>() {
-                            private int count = 0;
-
-                            @Override
-                            public Socket answer(InvocationOnMock invocationOnMock)
-                                    throws Throwable {
-                                if (count < 2) {
-                                    count++;
-                                    throw new SocketTimeoutException();
-                                }
-
-                                return new Socket();
-                            }
-                        });
-
-        assertNotNull(NetUtils.acceptWithoutTimeout(serverSocket));
+        Socket expected = new Socket();
+        ServerSocket serverSocket =
+                new ServerSocket() {
+                    private int count = 0;
+
+                    @Override
+                    public Socket accept() throws IOException {
+                        if (count < 2) {
+                            count++;
+                            throw new SocketTimeoutException();
+                        }
+
+                        return expected;
+                    }
+                };
+
+        assertEquals(expected, NetUtils.acceptWithoutTimeout(serverSocket));
+
+        // Validates timeout option precondition
+        serverSocket =
+                new ServerSocket() {
+                    @Override
+                    public Socket accept() throws IOException {
+                        return expected;
+                    }
+                };
+
+        // non-zero timeout (throw exception)
```

Comment:
let's add a separate test for this. It can then be as simple as:


```
@Test(expected = IllegalArgumentException.class)
public void testAcceptWithoutTimeoutRejectsSlotWithSoTimeout() {
    try (final ServerSocket serverSocket = new ServerSocket(0)) {
        serverSocket.setSoTimeout(5);
    
        NetUtils.acceptWithoutTimeout(serverSocket);
    }
}
```